packages/core/src/util/ledger.ts

```diff
@@ -0,0 +1,26 @@
+import Transport from "@ledgerhq/hw-transport";
+import TransportWebHID from "@ledgerhq/hw-transport-webhid";
+
+async function getDevicePaths(ledgerLibrary: typeof TransportWebHID): Promise<ReadonlyArray<HIDDevice>> {
+  const supported = await ledgerLibrary.isSupported();
+  if (!supported) {
+    throw new Error("Your computer does not support the ledger!");
+  }
+  return await ledgerLibrary.list();
+}
+
+async function getWebHIDTransport(): Promise<Transport> {
+  try {
+    return await TransportWebHID.create();
+  } catch (error) {
+    if ((error as Error).message !== "The device is already open.") throw error;
+
+    const devices = await getDevicePaths(TransportWebHID);
+    const transport = new TransportWebHID(devices[0]);
+    return transport as Transport;
+  }
+}

```

Comment:
we'll be creating a separate package for ledger functionalities.
can we move this to `polynetwork` package first?